apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -727,30 +730,60 @@ where
     /// If a handle to an Ethereum oracle was provided to the [`Shell`], attempt
     /// to signal it to start, using an initial configuration based on
     /// Ethereum bridge parameters in blockchain storage.
-    fn ensure_ethereum_oracle_started(&mut self) {
+    ///
+    /// This method must be safe to call even before ABCI `InitChain` has been
+    /// called (i.e. when storage is empty), as we may want to do this check
+    /// every time the shell starts up (including the first time ever at which
+    /// time storage will be empty).
+    fn start_ethereum_oracle_if_necessary(&mut self) {
         if let ShellMode::Validator {
             eth_oracle: Some(EthereumOracleChannels { control_sender, .. }),
             eth_oracle_started,
             ..
         } = &mut self.mode
         {
             if *eth_oracle_started {
+                tracing::info!("Not starting oracle as it was already started");
+                return;
+            }
+            // We *always* expect a value describing the status of the Ethereum
+            // bridge to be present under [`eth_bridge::storage::active_key`],
+            // once a chain has been initialized. We need to explicitly check if
+            // this key is present here because we may be starting up the shell
+            // for the first time ever, in which case the chain hasn't been
+            // initialized yet.
+            let (has_key, _) = self
```

Comment:
I've added a comment/some extra context to the fn docstring, to make clear why we need to do this explicit `has_key` check - this fn should be valid to call even before `InitChain` has been called (i.e. the first time a validator node ever starts up)